### ISSUE TITLE
🐛 [Bug] Donations slider broken when missing translations

### ISSUE THREAD

**Issue by mattwr18** [Human-Connection/Human-Connection#2356]
## :bug: Bugreport

If you are using a language that has untranslated phrases/words for the donations section, it doesn't show the amount or any fallback language like English

### Steps to reproduce the behavior
1. log into your account
2. switch to Italian
3. see that the Donations slider is broken
4. ...
5. Profit

### Expected behavior

It is quite important that this information is always displayed, even if it's in a fallback language

![Screenshot from 2019-11-25 19-24-08](https://user-images.githubusercontent.com/26943915/69567376-87cdcf80-0fb9-11ea-9e0e-5360ebf04e9a.png)

**Comment by erhardtstiefel** [comment]
Im an absolutly newcomer and this is my first contribution - if possible. I can reproduce this behaviour frontend but where can I see the code?

**Comment by mattwr18** [comment]
oh, hey @erhardtstiefel I didn't see this until now.. sorry.. I think you found where the translations go, no?
I guess this will be a problem for any missing translations, so I think we need to have a backup, like English
Here is a screen shot of more broken code on production 
![Screenshot from 2019-12-13 10-08-28](https://user-images.githubusercontent.com/26943915/70788274-cffa3980-1d90-11ea-9c40-df81557c4d4c.png)

**Comment by erhardtstiefel** [comment]
Hello @mattwr18 , yes this is the first issue I picked up myself to start with, but @roschaefer had other plans to bring me in a shape where I could contribute to this fascinating project. As soon as possible I'll look now at this issue and if I've problems I'll ask you. Ok?

**Comment by mattwr18** [comment]
great, some more feedback for you is that it looks like the only one it doesn't work for is Italian
all others have a fallback of English when the translation is missing... perhaps Italian isn't registered properly?

**Comment by erhardtstiefel** [comment]
I added the missing italian verbs just for the display of "donations" but I see a lot more verbs missing in it.json. Does it make sense to translate and  fill in everything which is to be found in en.json ?

**Comment by mattwr18** [comment]
That would be awesome, yes @erhardtstiefel

**Comment by erhardtstiefel** [comment]
@mattwr18 Can I use this issue as a reference for a PR that contains all my additions to it.json or do I have to create a new issue?

**Comment by erhardtstiefel** [comment]
1.) I have added all missing texts in it.json. Therefore I translated the texts from en.json with DeepL into Italian. 
2.) How can I test my work?
3.) I found a difference in "moderation": { "name": "Moderation", "reports": { ....
between en.json and it.json. It looks like there was code added in en.json but forgotten to syncronise it to it.json. What can I do?

**Comment by erhardtstiefel** [comment]
Which type of activity do you else expect? I asked some questions and didn't get any answer.

**Comment by rbeer** [comment]
Hello Erhard! :)

> Can I use this issue as a reference for a PR that contains all my additions to it.json or do I have to create a new issue?

Yes, just reference this one.

> 2.) How can I test my work?
You will have to get a local development instance running. THere is a [guide](https://docs.human-connection.org/human-connection/installation) for that.

> 3.) I found a difference in "moderation": { "name": "Moderation", "reports": { ....
between en.json and it.json. It looks like there was code added in en.json but forgotten to syncronise it to it.json. What can I do?

Since you have added the other translations, you could do the same with those missing and add them.

Having translations is obviously the better choice over fallback strings, but let me address the problem.

**tl;dr** _i18n finds empty strings in locale it.json and uses them. The defined fallback language 'en' isn't used on empty strings, but only on `undefined`, i.e. when the key is missing._

The translation flow is:

0. (during setup): define `fallbackLanguage` to be 'en'
1. figure out `language` to use ('de', 'en', 'it', ...)
2. look up `translatedString` for `key` in `language`
3. _is `translatedString` defined?_
3.1 _yes_ - use it
3.2 _no_ - use `translatedString` from `fallbackLanguage`

Which explains the missing keys in 'it'.
'de' is the only file that has the same amount of lines - implying amount of `key`s - as 'en'.
![locale-lines](https://user-images.githubusercontent.com/3411649/79637086-8a091400-817c-11ea-9ee7-09c529eb0f47.png)

What makes 'it' different and causes this bug is that 'it' is the only language that has empty strings as `translatedString`s.
![emptyString-it-only](https://user-images.githubusercontent.com/3411649/79637268-46170e80-817e-11ea-91db-6e09b3003350.png)

The empty strings then get used, instead of replaced by the value from `fallbackLanguage`. Leaving empty containers behind, that sometimes inflate completely:
![login_en](https://user-images.githubusercontent.com/3411649/79637437-f0436600-817f-11ea-98d5-da3e1f5a8abc.png)
![login_it](https://user-images.githubusercontent.com/3411649/79637435-efaacf80-817f-11ea-9bfc-c23306608e91.png)

Takeaway: Don't use empty strings! Either add a translation or leave the key undefined to use the fallback language (i.e. 'en'glish)

**Comment by rbeer** [comment]
Not fixed
https://github.com/Human-Connection/Human-Connection/blob/32e5801d20ddcb3c56407dc7f02f00c8c316337a/webapp/locales/it.json#L36-L40

**Comment by erhardtstiefel** [comment]
Sorry, but as I mentioned I do not know how to fix it although I'm interested to know how to do it.

**Comment by mattwr18** [comment]
I think this issue can be closed @rbeer and @erhardtstiefel because the issue is about the Donations slider that is no longer visible for users, nor is it clear if it will ever be used again.
There might be another issue that there are missing translations, but as far as I remember @roschaefer made to where there is a fallback language, which is English, I believe. Obviously, we would like to have everything translated in Italian, but that relies on Italian language users adding them in lokalise or in the code base.

**Comment by rbeer** [comment]
@mattwr18 Maybe rename it, because the issue is not minor, yet easy to fix.
The missing (or rather empty) strings often let their component's layout collapse; entire buttons, links, texts just disappearing.

@erhardtstiefel 
You basically already did the major bulk of the work.
Take a look at 9ec166e! Simply removing all those keys that have empty strings as values solves the issue, because then the fallback language is used.
Since you have already translated many of the missing keys, your version of it.json is obviously preferred. So what you would have to do is just remove all the empty strings you have left and open a pull request for that one file (it.json).

**Comment by erhardtstiefel** [comment]
Scusate, non sono italiano e non sono un programmatore. Volevo solo aiutare a modificare un numero che è stato dichiarato per i principianti.
Sorry, I'm not Italian and I'm not a programmer. I just wanted to help editing an issue that was declared for beginners.
Tut mir leid, ich bin kein Italiener und kein Programmierer. Ich wollte nur helfen ein issue zu bearbeiten, das für Anfänger deklariert war.

**Comment by erhardtstiefel** [comment]
Hi, anybody still ongoing with Human Connection?
